src/go/configmanager/config_manager.go

```diff
@@ -249,13 +251,13 @@ func (m *ConfigManager) applyServiceConfig(serviceConfig *confpb.Service) error
 	if err != nil {
 		return fmt.Errorf("fail to make a snapshot, %s", err)
 	}
-	return m.cache.SetSnapshot(m.envoyConfigOptions.Node, *snapshot)
+	return m.cache.SetSnapshot(context.Background(), m.envoyConfigOptions.Node, *snapshot)
 }
 
 func (m *ConfigManager) makeSnapshot() (*cache.Snapshot, error) {
 	m.Infof("making configuration for api: %v", m.serviceInfo.Name)
 
-	var clusterResources, endpoints, secrets, runtimes, routes, listenerResources []types.Resource
+	var clusterResources, listenerResources []types.Resource
```

Comment:
Where are the other resources, like routes?

Reply:
They are empty,  we don't generate other resources beside listeners and clousters.   I believe we don't need to set empty resources to NewSnapshot.